Src/FluentAssertions/Numeric/NumericAssertions.cs

```diff
@@ -481,5 +481,53 @@ public AndConstraint<TAssertions> NotBeOfType(Type unexpectedType, string becaus
             throw new NotSupportedException("Calling Equals on Assertion classes is not supported.");
 
         private protected virtual bool IsNaN(T value) => false;
+
+        /// <summary>
+        /// A method to generate additional information upon comparison failures.
+        /// </summary>
+        /// <param name="expected">The value to compare the current numeric value with.</param>
+        /// <returns>
+        /// Returns the difference between the integral number value and the <paramref name="expected" /> value.
+        /// </returns>
+        private protected abstract T? CalculateDifferenceForFailureMessage(T expected);
+
+        /// <summary>
+        /// A method to determine what is the minimal difference threshold from which to generate a "difference failure message".
+        /// </summary>
+        /// <returns>
+        /// The minimal difference threshold from which to generate a failure message.
+        /// </returns>
+        private protected abstract T GetMinimalDifferenceThresholdForFailureMessage();
+
+        /// <summary>
+        /// A method to determine what is the maximal difference value from which to generate a "difference failure message".
+        /// </summary>
+        /// <returns>
+        /// The maximal difference value from which to generate a failure message.
+        /// </returns>
+        private protected abstract T GetMaximalDifferenceThresholdForFailureMessage();
+
+        private string GenerateDifferenceMessage(T? expected)
+        {
+            const string noDifferenceMessage = ".";
+            if (!Subject.HasValue || !expected.HasValue)
+            {
+                return noDifferenceMessage;
+            }
+
+            var difference = CalculateDifferenceForFailureMessage(expected.Value);
+            if (difference is null)
+            {
+                return noDifferenceMessage;
+            }
+
+            if (difference.Value.CompareTo(GetMinimalDifferenceThresholdForFailureMessage()) < 0 ||
```

Comment:
🤔 I think you've misunderstood me. What I meant is that if the subject and expectation are both under 10, you don't need FA to show you the difference. 
🤔 I wonder why you decided to introduce so much complexity if the `CalculateDifferenceForFailureMessage` can determine itself whether it's worth to return a difference for small numbers or return `null`.

Reply:
I wanted a single location to decide whether or not to generate the "difference message", which logically should be in the `GenerateDifferenceMessage` method.
Unfortunately, I could not simply run the logic on the result that is returned from `CalculateDifferenceForFailureMessage`, as the result is a `T` on which I cannot perform `<` and `>` operations.
Therefore I went with the approach of comparing the `T` difference with concrete values returned from the Minimal/Maximal methods.

Anyway, your suggestion is a lot simpler.
I'll add logic to specific `CalculateDifferenceForFailureMessage` methods to return `null` if we `subject` and `expectation` are under 10 (int, uint, short, ushort, long, ulong).
And for the rest (decimal, float, double, byte, sbyte) we will always return the message.
WDYT?

Reply:
Yeah, sounds like a good idea.

Reply:
Yes, I agree. But I can imagine that when you have large numbers like 12345784, a difference of 3 might be less obvious to the eye.